### PR TITLE
Add dark mode support for image documents

### PR DESCRIPTION
**Pull request by ghost**
https://bugs.webkit.org/show_bug.cgi?id=243204

Reviewed by NOBODY (OOPS!).

* Source/WebCore/html/ImageDocument.cpp:
(WebCore::ImageDocument::createDocumentStructure):

### CODE REVIEW

**Comment by pxlcoder** [comment]
It looks like the PR description is incorrectly formatted, did you use `git webkit pr`? Instructions can be found [here](https://github.com/WebKit/WebKit/wiki/Pull-Requests).

File: Source/WebCore/html/ImageDocument.cpp
Comment (pxlcoder) [inline Source/WebCore/html/ImageDocument.cpp]: Nit: Can we use `light dark` instead, to match the other instances of this string in the project?

Comment (pxlcoder) [inline Source/WebCore/html/ImageDocument.cpp]: Nit: Please move this up after the element creation (L222).

**Review by pxlcoder — COMMENTED** [review]
Can we add a test for the new behavior?

**Comment by ghost** [comment]
> It looks like the PR description is incorrectly formatted, did you use `git webkit pr`? Instructions can be found [here](https://github.com/WebKit/WebKit/wiki/Pull-Requests).

I tried to use this tool, but it failed horribly severall times. I assumed that I was supposed to make the PR manually *shrug* the template only had some vague instructions.
(The tool failed to create a fork of webkit/webkit, instead renaming my fork of haiku/haikuwebkit... it didnt manage to remember the token or account either, and completely stalled every time after asking me whether I want to make a new fork, regardless of weather i said yes or no)

Why does the description matter? Isn't the changelog stuff taken from the commit?

As for a test, sure, I am a bit unsure on the how though, I've only written a test for a layout of the ftp directory code yet. I assume this would be an image comparison based one?

**Comment by ghost** [comment]
(also, what does Nit mean?)

**Comment by pxlcoder** [comment]
> > It looks like the PR description is incorrectly formatted, did you use `git webkit pr`? Instructions can be found [here](https://github.com/WebKit/WebKit/wiki/Pull-Requests).
> 
> I tried to use this tool, but it failed horribly severall times. I assumed that I was supposed to make the PR manually _shrug_ the template only had some vague instructions. (The tool failed to create a fork of webkit/webkit, instead renaming my fork of haiku/haikuwebkit... it didnt manage to remember the token or account either, and completely stalled every time after asking me whether I want to make a new fork, regardless of weather i said yes or no)

It might be good to file a bug or raise this issue in the #webkit Slack.
 
> Why does the description matter? Isn't the changelog stuff taken from the commit?

Not entirely sure, but I'm hoping the description will just get cleaned up after merging, so this is probably fine for now.
 
> As for a test, sure, I am a bit unsure on the how though, I've only written a test for a layout of the ftp directory code yet. I assume this would be an image comparison based one?

I think you can add a reference mismatch test, like the one in https://commits.webkit.org/252673@main. Basically, you would have two html files (`.html` and `-expected-mismatch.html`) and the test will pass if the snapshots of those pages are different.

**Comment by pxlcoder** [comment]
> (also, what does Nit mean?)

Short for "nitpick".

**Comment by ghost** [comment]
I've added the test, based on how it was done in the text/plain change.
One question I have about this is wether it is expected that the testrunner *always* runs with a light mode, does it force this normally? or is this just coincidentally what it currently does "normally"?

If it is nondeterminate it might make sense to add a force to the light mode to the other html test file aswell.

**Comment by pxlcoder** [comment]
> I've added the test, based on how it was done in the text/plain change. One question I have about this is wether it is expected that the testrunner _always_ runs with a light mode, does it force this normally? or is this just coincidentally what it currently does "normally"?
> 
> If it is nondeterminate it might make sense to add a force to the light mode to the other html test file aswell.

Yes, tests are always run in a light mode environment. `internals.settings.setUseDarkAppearance` simply simulates a dark mode environment, it does not actually change any OS level settings.

**Review by pxlcoder — APPROVED** [review]

**Comment by ghost** [comment]
Hey, do I need to do something more to make this ready to commit? The doc sais something about adding a merge label, but atleast on the github mobile page I see no way to do this.

**Comment by pxlcoder** [comment]
You need [committer access](https://webkit.org/commit-and-review-policy/) to merge, which requires ~10 commits and a nomination.

I was waiting on tests to finish running, and they have, so I will merge now 😄

Thank you for the PR!

**Comment by ghost** [comment]
Looks like I broke the merge tool. :(

**Comment by pxlcoder** [comment]
I think you will need to use `git webkit setup` and `git webkit pr` to make the pull request.

It looks like your WebKit is forked from https://github.com/haiku/haikuwebkit rather than this repository?

**Comment by ghost** [comment]
Yes it is,
webkit-git setup renamed my existing repository instead of forking the webkit/webkit repo.

I don't see how I can fix this.
I suppose I can just upload it to bugzilla next week if the tool is still broken then?

**Comment by pxlcoder** [comment]
I see. I think you will need to clone *this* repo (https://github.com/WebKit/WebKit) and then run `git webkit setup` from inside there. Or is the issue that you cloned this repo, but the setup script still went to your other fork?

Bugzilla is also fine, but would be good to get more information on what is failing, so that we can address the root issue.

**Comment by ghost** [comment]
> I see. I think you will need to clone _this_ repo (https://github.com/WebKit/WebKit) and then run `git webkit setup` from inside there. Or is the issue that you cloned this repo, but the setup script still went to your other fork?
> 
> Bugzilla is also fine, but would be good to get more information on what is failing, so that we can address the root issue.

Yep, I had both haikuwebkit and webkit cloned, and used the script in the webkit repo.

**Comment by ghost** [comment]
The failure in the pull request log thing is that it is using the wrong repo url anyhow, it renamed the repo but the uri is still haikuwebkit.git and not webkit.git at the end for the repo it tried to pull from

**Comment by pxlcoder** [comment]
@aj062 @JonWBedard Any ideas what might be going on with the setup script in @nephele-gh's case?

**Comment by pxlcoder** [comment]
@nephele-gh Feel free to upload the same patch on Bugzilla, whenever you are able to.

**Comment by ghost** [comment]
I would have time now to upload it to bugzilla, I kind of forgot the process, and the doc has been replaced with the git one :D.

Maybe the best path forward would be to figure out why the merge and setup scripts broke and fix those.

**Comment by pxlcoder** [comment]
> I would have time now to upload it to bugzilla, I kind of forgot the process, and the doc has been replaced with the git one :D.

Running `webkit-patch upload -g HEAD 243204` from your branch should upload the patch to Bugzilla.

> Maybe the best path forward would be to figure out why the merge and setup scripts broke and fix those.

@aj062 Is this something you can help out with?

**Pull request closed without merging**
